resolwe/observers/consumers.py

```diff
@@ -0,0 +1,70 @@
+"""Consumers for Observers."""
+
+from channels.generic.websocket import JsonWebsocketConsumer
+
+from django.contrib.contenttypes.models import ContentType
+
+from .models import Observer, Subscription
+from .protocol import CHANGE_TYPE_DELETE, GROUP_SESSIONS
+
+
+class ClientConsumer(JsonWebsocketConsumer):
+    """Consumer for client communication."""
+
+    def websocket_connect(self, message):
+        """Handle establishing a WebSocket connection."""
+        session_id = self.scope["url_route"]["kwargs"]["session_id"]
+        self.session_id = session_id
```

Comment:
Is there one consumer instance per client connection or one consumer instance for all clients?

If the second is true, then `self.session_id` (which is client-specific) will always point to the session id of the client that performed the last connection.

Reply:
The first is true.